### ISSUE TITLE
Error_Packet "partial packet: expecting a bytes, got: b"

### ISSUE THREAD

**Issue by dulguun0225** [commercialhaskell/stack#3800]
On Windows build v1709, WSL. Also other builds and installs fail with `Error_Packet "partial packet: expecting a bytes, got: b"` error.
```
stack build lens
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
Downloading snapshot
Downloading mirrors
Cannot update index (no local copy)
Downloading index
Unable to load cabal files for snapshot

----
Deleting cached snapshot file: /home/d/.stack/build-plan/lts-10.3.yaml
Recommendation: try running again. If this fails again, open an upstream issue at:
https://github.com/fpco/lts-haskell/issues/new
----

Updating package index Hackage (mirrored at https://s3.amazonaws.com/hackage.fpcomplete.com/) ...Error_Packet "partial packet: expecting 16384 bytes, got: 9605"
```

**Comment by dulguun0225** [comment]
I've tried it on native windows stack. `stack setup` randomly fails with `Error_Packet`.

**Comment by mgsloan** [comment]
Curious!  I wonder if this is an issue with hackage-security or with S3.  I'm guessing it's an upstream problem with hackage-security, or at least with error reporting.

**Comment by ocheron** [comment]
This looks like a type of issue fixed in tls-1.4.0 and 1.4.1.
Most likely the one described in snoyberg/http-client#273.
